### ISSUE TITLE
Crash loading Yuki-FM homebrew (Lua Player based)

### ISSUE THREAD

**Issue by voltagex** [citra-emu/citra#2519]
- You have provided the version (commit hash) of Citra you are using.
d0bf7df

- You have provided sufficient detail for the issue to be reproduced.
1. git clone https://github.com/coc4tm/Yuki-FM

2. chmod +x the Linux build tools in the repo

3. `make`

4. Run `build/Yuki FM/Yuki FM.3dsx`

- You have provided system specs (if relevant).
Fedora 25 using a nightly binary from Citra site

- Please also provide:
  - For crashes, a backtrace.
```
[voltagex@badger Yuki FM]$ ~/Downloads/citra-linux-20170131-d0bf7df/citra Yuki\ FM.3dsx 
[   0.000000] Config <Info> citra/config.cpp:LoadINI:36: Successfully loaded /home/voltagex/.config/citra-emu/sdl2-config.ini
[   0.005169] Audio <Error> audio_core/sink_details.cpp:GetSinkDetails:32: AudioCore::SelectSink given invalid sink_id 
[   0.140395] Loader <Info> core/loader/loader.cpp:GetLoader:142: Loading file Yuki FM.3dsx as 3DSX...
[   0.151050] Service.FS <Info> core/file_sys/archive_sdmc.cpp:ArchiveFactory_SDMC:309: Directory /home/voltagex/.local/share/citra-emu/sdmc/ set as SDMC.
[   0.151161] Common.Filesystem <Warning> common/file_util.cpp:CreateFullPath:200: path exists /home/voltagex/.local/share/citra-emu/sdmc/
[   0.151217] Service.FS <Info> core/file_sys/archive_sdmcwriteonly.cpp:ArchiveFactory_SDMCWriteOnly:35: Directory /home/voltagex/.local/share/citra-emu/sdmc/ set as SDMCWriteOnly.
[   0.151241] Common.Filesystem <Warning> common/file_util.cpp:CreateFullPath:200: path exists /home/voltagex/.local/share/citra-emu/sdmc/
[   0.151266] Service.FS <Info> core/file_sys/archive_source_sd_savedata.cpp:ArchiveSource_SDSaveData:42: Directory /home/voltagex/.local/share/citra-emu/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/title/ set as SaveData.
[   0.151303] Service.FS <Info> core/file_sys/archive_extsavedata.cpp:ArchiveFactory_ExtSaveData:176: Directory /home/voltagex/.local/share/citra-emu/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/ set as base for ExtSaveData.
[   0.151331] Common.Filesystem <Warning> common/file_util.cpp:CreateFullPath:200: path exists /home/voltagex/.local/share/citra-emu/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/
[   0.151354] Service.FS <Info> core/file_sys/archive_extsavedata.cpp:ArchiveFactory_ExtSaveData:176: Directory /home/voltagex/.local/share/citra-emu/nand/data/00000000000000000000000000000000/extdata/ set as base for ExtSaveData.
[   0.151378] Common.Filesystem <Warning> common/file_util.cpp:CreateFullPath:200: path exists /home/voltagex/.local/share/citra-emu/nand/data/00000000000000000000000000000000/extdata/
[   0.151778] Service.APT <Warning> core/hle/service/apt/apt.cpp:Init:524: Unable to load shared font: /home/voltagex/.local/share/citra-emu/sysdata/shared_font.bin
[   0.153482] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:495: GL_VERSION: 4.5 (Core Profile) Mesa 13.0.3
[   0.153508] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:496: GL_VENDOR: Intel Open Source Technology Center
[   0.153518] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:497: GL_RENDERER: Mesa DRI Intel(R) Iris Graphics 540 (Skylake GT3e) 
[   0.183180] Service.FS <Error> core/file_sys/archive_romfs.cpp:ArchiveFactory_RomFS:20: Unable to read RomFS!
[   0.238997] Service.SRV <Warning> core/hle/service/srv.cpp:RegisterClient:40: (STUBBED) called
[   0.253572] Service.APT <Warning> core/hle/service/apt/apt.cpp:GetLockHandle:120: (STUBBED) called handle=0x00018004 applet_attributes=0x00000000
[   0.261814] Service.APT <Warning> core/hle/service/apt/apt.cpp:Enable:128: (STUBBED) called attributes=0x00000000
[   0.263412] Service.APT <Warning> core/hle/service/apt/apt.cpp:NotifyToWait:102: (STUBBED) app_id=768
[   0.298490] Service.APT <Warning> core/hle/service/apt/apt.cpp:ReceiveParameter:224: called app_id=0x00000300, buffer_size=0x00001000
[   0.305455] Service <Error> core/hle/service/service.cpp:HandleSyncRequest:95: unknown / unimplemented function 'ConfigureNew3DSCPU': port=ptm:sysm, cmd_buff[1]=0x0
[   0.329805] Service.PTM <Warning> core/hle/service/ptm/ptm.cpp:CheckNew3DS:107: (STUBBED) called isNew3DS = 0x00000000
[   0.329827] Service.APT <Warning> core/hle/service/apt/apt.cpp:CheckNew3DS:493: (STUBBED) called
[   0.377609] Service.GSP <Warning> core/hle/service/gsp_gpu.cpp:AcquireRight:690: called
[   0.379452] Service.GSP <Warning> core/hle/service/gsp_gpu.cpp:RegisterInterruptRelayQueue:384: called, flags=0x00000001
[   0.402688] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:318: (STUBBED) called command=0x00000007, buffer1_size=0x00000004, buffer2_size=0x00000001, buffer1_addr=0x08009410, buffer2_addr=0x08009414
[   0.404641] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:318: (STUBBED) called command=0x00000004, buffer1_size=0x00000001, buffer2_size=0x00000001, buffer1_addr=0x08009410, buffer2_addr=0x08009414
[   0.405406] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:318: (STUBBED) called command=0x00000004, buffer1_size=0x00000001, buffer2_size=0x00000001, buffer1_addr=0x08009410, buffer2_addr=0x08009414
[   0.432817] Kernel.SVC <Warning> core/hle/svc.cpp:CreateMemoryBlock:943: called addr=0x08012000
[   0.433686] Service <Error> core/hle/service/service.cpp:HandleSyncRequest:95: unknown / unimplemented function 'Initialize': port=http:C, cmd_buff[1]=0x1000, cmd_buff[2]=0x20, cmd_buff[3]=0xF802A, cmd_buff[4]=0x0, cmd_buff[5]=0x10002B
[   0.452342] Service <Error> core/hle/service/service.cpp:HandleSyncRequest:95: unknown / unimplemented function 'ControlArchive': port=fs:USER, cmd_buff[1]=0x5, cmd_buff[2]=0x0, cmd_buff[3]=0x1, cmd_buff[4]=0xC, cmd_buff[5]=0x8, cmd_buff[6]=0xCA, cmd_buff[7]=0x80075DC, cmd_buff[8]=0x8C, cmd_buff[9]=0x8007588
[   0.462778] Service.FS <Error> core/hle/service/fs/archive.cpp:HandleSyncRequest:110: Reading from out of bounds offset=0x0 length=0x00000028 file_size=0x0
Segmentation fault (core dumped)

```

**Comment by mailwl** [comment]
the problem in https://github.com/citra-emu/citra/blob/master/src/core/loader/3dsx.cpp#L299
different header struct and size in header

**Comment by wwylele** [comment]
@mailwl what, shouldn't that prevent the app from being even longing?

**Comment by voltagex** [comment]
@mailwl / @wwylele is this simple to fix? Is it a Citra issue or a problem with the homebrew code?

**Comment by mailwl** [comment]
to fix - simple, just delete lines 299 and 300 from code to remove condition, but it's workaround only. Not really fix the problem

**Comment by wwylele** [comment]
I don't think what @mailwl said is correct. According to [3dsx format](https://www.3dbrew.org/wiki/3DSX_Format), that size field indicates whether the ROM contains an extended header (i.e. contains RomFS). The ROM Yuki-FM using is a non-RomFS version of lpp-3ds. According to lpp-3ds source code, it tries to read the script from RomFS; if it fails, it then fallback to SD file reading. The problem in citra is, if a game tries to read non-exist RomFS, it will simply crash instead of returning a proper error code.

**Comment by wwylele** [comment]
Fixed by #2561
